libraries/chain/include/eosio/chain/wasm_interface_private.hpp

```diff
@@ -164,7 +164,11 @@ struct eosvmoc_tier {
             }
          }
 #endif
-         const bool allow_oc_interrupt = attempt_tierup && context.is_applying_block() && context.trx_context.has_undo();
+         // do not allow oc interrupt if no undo as the transaction needs to be undone to restart it.
+         // do not allow oc interrupt if implicit as deferred trx onerror execute as implicit outside the transaction
+         // retry in transaction_context. Also scheduled manage the undo stack explicitly.
+         const bool allow_oc_interrupt = attempt_tierup && context.is_applying_block() &&
+                                         context.trx_context.has_undo() && !context.trx_context.is_implicit() && !context.trx_context.is_scheduled();
```

Comment:
Do we need both `!context.trx_context.is_implicit()` and `!context.trx_context.is_scheduled()`? Is `!context.trx_context.is_scheduled()` sufficient?